sdk/python/feast/infra/materialization/snowflake_engine.py

```diff
@@ -271,32 +271,68 @@ def _materialize_one(
+                    unique_entities = '"' + '", "'.join(join_keys) + '"'
+
+                    query = f"""
+                    SELECT
+                        COUNT(*)
+                    FROM
+                    (
+                        SELECT
+                            {unique_entities}
+                        FROM
+                            {feature_view.batch_source.get_table_query_string()}
+                        QUALIFY ROW_NUMBER() OVER (PARTITION BY {unique_entities} ORDER BY {unique_entities}) = 1
+                    )
+                    """
+
+                    with get_snowflake_conn(self.repo_config.offline_store) as conn:
+                        rows_to_write = conn.cursor().execute(query).fetchall()[0][0]
+            else:
+                rows_to_write = 1  # entityless feature view has a placeholder entity
+
             if feature_view.batch_source.field_mapping is not None:
                 fv_latest_mapped_values_sql = _run_snowflake_field_mapping(
                     fv_latest_values_sql, feature_view.batch_source.field_mapping
                 )
 
-            fv_to_proto_sql = self.generate_snowflake_materialization_query(
-                self.repo_config,
-                fv_latest_mapped_values_sql,
-                feature_view,
-                project,
-            )
+            features_full_list = feature_view.features
+            feature_batches = [
+                features_full_list[i : i + 100]
+                for i in range(0, len(features_full_list), 100)
+            ]
 
-            if self.repo_config.online_store.type == "snowflake.online":
-                self.materialize_to_snowflake_online_store(
+            for i, feature_batch in enumerate(feature_batches):
+                fv_to_proto_sql = self.generate_snowflake_materialization_query(
                     self.repo_config,
-                    fv_to_proto_sql,
+                    fv_latest_mapped_values_sql,
                     feature_view,
+                    feature_batch,
                     project,
                 )
-            else:
-                self.materialize_to_external_online_store(
-                    self.repo_config,
-                    fv_to_proto_sql,
-                    feature_view,
-                    tqdm_builder,
-                )
+
+                if self.repo_config.online_store.type == "snowflake.online":
+                    self.materialize_to_snowflake_online_store(
+                        self.repo_config,
+                        fv_to_proto_sql,
+                        feature_view,
+                        project,
+                    )
+                else:
+                    click.echo(
+                        f"Materializing batch {Style.BRIGHT + Fore.GREEN}#{i+1}{Style.RESET_ALL} of {Style.BRIGHT + Fore.GREEN}{len(feature_batch)}{Style.RESET_ALL} features"
+                    )
+                    with tqdm_builder(rows_to_write) as pbar:
```

Comment:
think here you need rows_to_write * len(feature_batches)

Reply:
@adchia I thought so to ... but apparently we aren't counting rows to write =  entities * features, we count just entities. Look at a simple "feast init" example.

Reply:
hmm in the old world that was correct before because we didn't split it up and there are only num_entities rows in a normal feast materialization. 

In this case, you're calling into the pbar more than that though since you also are doing batches. so i think you'd hit 100% after the first batch of 100 features, and show nothing on the second batch onwards?

Reply:
@adchia it should reset every loop of 100, no? the way you are suggesting would give the feel of one bar that shows full completion but the final number wouldnt make any sense

Reply:
@adchia updated to one bar ...merge it